From 24db9031d019201d90accedaa3f73a66e18009f6 Mon Sep 17 00:00:00 2001
From: Jianing Ren <jianing.ren@rock-chips.com>
Date: Thu, 28 Nov 2019 09:18:15 +0800
Subject: [PATCH] phy: phy-rockchip-inno-usb2: add iddig_output/en regs for
 rk3399

In order to switch the drd mode of the dwc3 controller by software, we
add these registers in the driver.

Change-Id: Ice5009631f491915ca78b5310cc9c623269bff41
Signed-off-by: Jianing Ren <jianing.ren@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-usb2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
index 6b9c644a6856..485bf726dd65 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
@@ -2613,6 +2613,8 @@ static const struct rockchip_usb2phy_cfg rk3399_phy_cfgs[] = {
 				.bvalid_det_clr	= { 0xe3d0, 3, 3, 0, 1 },
 				.bypass_dm_en   = { 0xe450, 2, 2, 0, 1 },
 				.bypass_sel     = { 0xe450, 3, 3, 0, 1 },
+				.iddig_output	= { 0xe454, 10, 10, 0, 1 },
+				.iddig_en	= { 0xe454, 9, 9, 0, 1 },
 				.idfall_det_en	= { 0xe3c0, 5, 5, 0, 1 },
 				.idfall_det_st	= { 0xe3e0, 5, 5, 0, 1 },
 				.idfall_det_clr	= { 0xe3d0, 5, 5, 0, 1 },
@@ -2661,6 +2663,8 @@ static const struct rockchip_usb2phy_cfg rk3399_phy_cfgs[] = {
 				.bvalid_det_en  = { 0xe3c0, 8, 8, 0, 1 },
 				.bvalid_det_st  = { 0xe3e0, 8, 8, 0, 1 },
 				.bvalid_det_clr = { 0xe3d0, 8, 8, 0, 1 },
+				.iddig_output	= { 0xe464, 10, 10, 0, 1 },
+				.iddig_en	= { 0xe464, 9, 9, 0, 1 },
 				.idfall_det_en	= { 0xe3c0, 10, 10, 0, 1 },
 				.idfall_det_st	= { 0xe3e0, 10, 10, 0, 1 },
 				.idfall_det_clr	= { 0xe3d0, 10, 10, 0, 1 },
-- 
2.35.3

